src/lib/forms/core/StringField.js

```diff
@@ -47,9 +47,11 @@ StringField.propTypes = {
   fieldPath: PropTypes.string.isRequired,
   inline: PropTypes.bool,
   optimized: PropTypes.bool,
+  width: PropTypes.number,
 };
 
 StringField.defaultProps = {
   inline: false,
   optimized: true,
+  width: null,
```

Comment:
`null` will shrink the String field to its default size which means if there are a few components in one row, they will not expand to cover it all.

<img width="1196" alt="Screenshot 2020-07-14 13 40 06" src="https://user-images.githubusercontent.com/230805/87416698-5a2ca980-c5cf-11ea-892b-20d76723dd4c.png">

We could use the maximum (16) number instead so it will `push` the rest of the elements to fill the row, and if its on its own it will fill the row.

<img width="1195" alt="Screenshot 2020-07-14 13 40 17" src="https://user-images.githubusercontent.com/230805/87416705-5bf66d00-c5cf-11ea-9b2b-4db6bc86b72d.png">

Reply:
yes you need to remember that this component is reused, and whatever you provide as a default will have the impact on all the other usages, which shouldn't be broken after your change